runtime/lib.rs

```diff
@@ -35,3 +35,46 @@ pub mod worker;
 
 mod worker_bootstrap;
 pub use worker_bootstrap::BootstrapOptions;
+
+// This is duplicated in `./build.rs`, keep in sync!
+deno_core::extension!(runtime,
+  deps = [
```

Comment:
`create_cli_snapshot()` didn't list `runtime` as an extension and had been implicitly depending on it from the built-upon snapshot, that needed to be added since we access to the extension's metadata across snapshots.

Turns out the `runtime` extensions was defined in `build.rs` and cli couldn't import it, I had to duplicate it.

Reply:
Could you explain the reasoning here a bit? Is it done just for the sake of ensuring extension ordering?

Reply:
This and the `ExtensionFileSourceCode::AlreadyEvaluated` stuff is so `ExtModuleLoader::resolve()` has access to the full list of extension module specifiers so it knows what to `Ok()` as is and what to defer to the inner module loader. I think I'm going to revert that part since it isn't necessary with the current approach. Actually we have no use for giving `ExtModuleLoader` access to the user's module loader as a fallback... we can just remove that and `Ok()` everything in `ExtModuleLoader::resolve()`.